crates/hyperdrive-math/src/lp.rs

```diff
@@ -224,6 +227,61 @@ impl State {
         present_value.into()
     }
 
+    /// NOTE: This version will not panic.
+    /// TODO: Combine this with `calculate_present_value`.
+    ///
+    /// Calculates the present value of LPs capital in the pool.
+    pub fn calculate_present_value_safe(
+        &self,
+        current_block_timestamp: U256,
+    ) -> Result<FixedPoint> {
+        let result = catch_unwind(AssertUnwindSafe(|| {
+            // Assume the following methods return Results and can be handled with `?`
+            let long_average_time_remaining = self.calculate_normalized_time_remaining(
+                self.long_average_maturity_time().into(),
+                current_block_timestamp,
+            );
+
+            let short_average_time_remaining = self.calculate_normalized_time_remaining(
+                self.short_average_maturity_time().into(),
+                current_block_timestamp,
+            );
+
+            // Conversion and calculation
+            let share_reserves = I256::try_from(self.share_reserves())
+                .map_err(|_| eyre!("Failed to convert share_reserves"))?;
+
+            let minimum_share_reserves = I256::try_from(self.minimum_share_reserves())
+                .map_err(|_| eyre!("Failed to convert minimum_share_reserves"))?;
+
+            let present_value = share_reserves
+                + self.calculate_net_curve_trade(
+                    long_average_time_remaining,
+                    short_average_time_remaining,
+                )
+                + self.calculate_net_flat_trade(
+                    long_average_time_remaining,
+                    short_average_time_remaining,
+                )
```

Comment:
The use of these non-safe functions means that this isn't "safe."

Reply:
thanks yeah you're right, i was just working on a totally safe version actually.  I'll do that in another PR here shortly.

Reply:
I have a PR up for this now, I'll update to use the new version when that PR lands.
https://github.com/delvtech/hyperdrive-rs/actions/runs/8897866606/job/24435373595?pr=11